### ISSUE TITLE
Cura changes printing direction when printing cylinder with multiple perimeters

### ISSUE THREAD

**Issue by radry** [Ultimaker/Cura#8495]
**Application version**
4.5 - 4.7.1

**Platform**
Windows 10 x64

**Printer**
Custom CoreXY

**Reproduction steps**
1. Create 100mm cylinder with wall thickness of 2x extrusion width or slightly more/less so that cura slices it with two outer perimeters only. For example in F360.
2. Load file into cura and slice, settings seemingly don't matter. Activate skirt with at least two perimeters to see different slicing behaviour.
3. Look at preview. See how skirt is printed and how the perimeters of the actual object are printed.

**Actual results**
Cura correctly creates two outer perimeters but changes the print direction at the start/finish of one perimeter. This causes an ugly Z seam which is impossible to get rid of.

**Expected results**
The perimeter walls continue printing in the same direction like it does on the skirt too.

**Project file**
[CFFFP_test.zip](https://github.com/Ultimaker/Cura/files/5323485/CFFFP_test.zip)

**Video**
[Preview](https://streamable.com/typ580)

**Comment by konskarm** [comment]
I think that the reason you are seeing this is that CuraEngine internally recognizes holes by the direction of the vertices in a polygon. The convention goes like this: if the vertices go counter-clockwise, then this means it is a solid object, while if they go clockwise then it is a hole. This is the reason you see the inner circle follow a clockwise direction (as it is internally used to identify that this is a hole), while the outer one having the opposite.

Now you do not see this happening in the skirt because the skirt is generated by offsetting the outer polygon, hence it keeps the CC direction and it is printed in the same order as the outer circle of the model.
If you use brim instead and you disable Brim Only on Outside, then the inner brim will be generated by offsetting the inner circle, thus again being printed in the other direction (similar to the inner circle of the model), so you see the same effect.

In short, this is expected behavior at the moment, but we have an internal ticket to change it.

I will close this as a duplicate of https://github.com/Ultimaker/Cura/issues/6210. Please post additional comments there.
